x-pack/solutions/observability/plugins/apm/server/routes/traces/get_unified_trace_items.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { APMEventClient } from '@kbn/apm-data-access-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { rangeQuery, termQuery } from '@kbn/observability-plugin/server';
+import type { Sort } from '@elastic/elasticsearch/lib/api/types';
+import { unflattenKnownApmEventFields } from '@kbn/apm-data-access-plugin/server/utils';
+import { MAX_ITEMS_PER_PAGE } from './get_trace_items';
+import {
+  PARENT_ID,
+  PROCESSOR_EVENT,
+  SPAN_DURATION,
+  SPAN_LINKS,
+  TRACE_ID,
+  TRANSACTION_DURATION,
+} from '../../../common/es_fields/apm';
+import type { APMConfig } from '../..';
+import { asMutableArray } from '../../../common/utils/as_mutable_array';
+import type { TraceItem } from '../../../common/waterfall/unified_trace_item';
+
+const fields = asMutableArray(['@timestamp', 'trace.id', 'service.name'] as const);
+
+const optionalFields = asMutableArray([
+  'span.id',
+  'span.name',
+  'duration',
+  'span.duration.us',
+  'transaction.duration.us',
+  'transaction.id',
+  'transaction.name',
+  'processor.event',
+  'parent.id',
+  'status.code',
+] as const);
```

Comment:
Could we use constants here?

Reply:
Don't we need the `OTEL_SPAN_LINKS_SPAN_ID` and `OTEL_SPAN_LINKS_TRACE_ID` fields here?

Reply:
I will not handle span links for now. I'll leave it for a follow up PR.